### PR TITLE
Add FAPI CIBA configuration

### PR DESCRIPTION
**Pull request by KalanaDananjaya**
### Proposed changes in this pull request

This PR introduces a new config to show support for FAPI profiles

Issues: 
https://github.com/wso2/product-is/issues/14178

#### General

- [x] **Is this PR explained thoroughly?** All code changes must be accounted for in the PR description.
- [x] **Is the PR labeled correctly?**

#### Functionality

- [x] **Are all requirements met?** Compare implemented functionality with the requirements specification.
- [x] **Does the UI work as expected?** There should be no Javascript errors in the console; all resources should load. There should be no unexpected errors. Deliberately try to break the feature to find out if there are corner cases that are not handled.

#### Code

- [x] **Do you fully understand the introduced changes to the code?** If not ask for clarification, it might uncover ways to solve a problem in a more elegant and efficient way.
- [x] **Does the PR introduce any inefficient database requests?** Use the debug server to check for duplicate requests.
- [x] **Are all necessary strings marked for translation?** All strings that are exposed to users via the UI must be [marked for translation](https://docs.djangoproject.com/en/1.10/topics/i18n/translation/).

#### Tests

- [ ] **Are there sufficient test cases?** Ensure that all components are tested individually; models, forms, and serializers should be tested in isolation even if a test for a view covers these components.
- [ ] **If this is a bug fix, are tests for the issue in place?**  There must be a test case for the bug to ensure the issue won’t regress. Make sure that the tests break without the new code to fix the issue.
- [ ] **If this is a new feature or a significant change to an existing feature?** has the manual testing spreadsheet been updated with instructions for manual testing?

#### Security

- [x] **Confirm this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets.**
- [x] **Are all UI and API inputs run through forms or serializers?**
- [x] **Are all external inputs validated and sanitized appropriately?**
- [ ] **Does all branching logic have a default case?**
- [x] **Does this solution handle outliers and edge cases gracefully?**
- [x] **Are all external communications secured and restricted to SSL?**

#### Documentation

- [ ] **Are changes to the UI documented in the platform docs?** If this PR introduces new platform site functionality or changes existing ones, the changes should be documented.
- [ ] **Are changes to the API documented in the API docs?** If this PR introduces new API functionality or changes existing ones, the changes must be documented.
- [ ] **Are reusable components documented?** If this PR introduces components that are relevant to other developers (for instance a mixin for a view or a generic form) they should be documented in the Wiki.

### CODE REVIEW

File: features/identity-core/org.wso2.carbon.identity.core.server.feature/resources/identity.xml [hunk 1]
```diff
@@ -537,6 +537,14 @@
             <!--AllowAdditionalParamsFromPostLogoutRedirectURI>false</AllowAdditionalParamsFromPostLogoutRedirectURI-->
         </OpenIDConnect>
 
+        <!--
+            FAPI Support.
+            Supported versions: IS 5.11.0 onwards
```
Comment (omindu) [inline features/identity-core/org.wso2.carbon.identity.core.server.feature/resources/identity.xml:542]: As this was not part of 5.11.0 GA, it's incorrect to say the support is from 5.11.0 onwards

File: features/identity-core/org.wso2.carbon.identity.core.server.feature/resources/identity.xml.j2 [hunk 2]
```diff
@@ -748,6 +748,18 @@
             {% endif %}
         </OpenIDConnect>
 
+         <!--
+            FAPI Support.
+            Supported versions: IS 5.11.0 onwards
+        -->
+        {% if fapi is defined %}
+        <FAPI>
```
Comment (mefarazath) [inline features/identity-core/org.wso2.carbon.identity.core.server.feature/resources/identity.xml.j2:756]: Can we rethink the configuration names?
Also wouldn't it be better to have this configuration under OpenIDConnect section?

File: features/identity-core/org.wso2.carbon.identity.core.server.feature/resources/identity.xml (hunk 1 above)
Comment (KalanaDananjaya) [reply features/identity-core/org.wso2.carbon.identity.core.server.feature/resources/identity.xml:542]: noted.will remove

File: features/identity-core/org.wso2.carbon.identity.core.server.feature/resources/identity.xml.j2 (hunk 2 above)
Comment (KalanaDananjaya) [reply features/identity-core/org.wso2.carbon.identity.core.server.feature/resources/identity.xml.j2:756]: Added this after a discussion with @VivekVinushanth .Do you have a suggestion? This is necessary in FAPI:CIBA conformance suite and FAPI security conformance suites to add an extra validation.

Comment (KalanaDananjaya) [reply features/identity-core/org.wso2.carbon.identity.core.server.feature/resources/identity.xml.j2:756]: How about something like <FapiSupported>true</FapiSupported> or <EnableFapi>true</EnableFapi>

Comment (KalanaDananjaya) [reply features/identity-core/org.wso2.carbon.identity.core.server.feature/resources/identity.xml.j2]: Changed to following after discussion
[fapi]
enable_ciba_profile=true
enable_security_profile=true

**Review by VivekVinushanth — APPROVED** [review]

**Review by omindu — APPROVED** [review]

File: features/identity-core/org.wso2.carbon.identity.core.server.feature/resources/identity.xml [hunk 3]
```diff
@@ -535,6 +535,14 @@
             Disable this config if additional params are not needed post logout redirect URL.
             -->
             <!--AllowAdditionalParamsFromPostLogoutRedirectURI>false</AllowAdditionalParamsFromPostLogoutRedirectURI-->
+
+            <!--
+                FAPI Profile Support.
+            -->
+            <!-- <FAPI>
+                <EnableCibaProfile>{{fapi.enable_ciba_profile}}</EnableCibaProfile>
+                <EnableSecurityProfile>{{fapi.enable_security_profile}}</EnableSecurityProfile>
```
Comment (senthalan) [inline features/identity-core/org.wso2.carbon.identity.core.server.feature/resources/identity.xml:544]: Why we need placeholders in the xml file

Comment (senthalan) [reply features/identity-core/org.wso2.carbon.identity.core.server.feature/resources/identity.xml:544]: we should have default values here.

Comment (KalanaDananjaya) [reply features/identity-core/org.wso2.carbon.identity.core.server.feature/resources/identity.xml]: Fixed

**Pull request merged**